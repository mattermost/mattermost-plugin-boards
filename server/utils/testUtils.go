// Copyright (c) 2020-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

package utils

import "github.com/stretchr/testify/mock"

var Anything = mock.MatchedBy(func(interface{}) bool { return true })
