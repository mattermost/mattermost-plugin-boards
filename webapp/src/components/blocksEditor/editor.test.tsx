// Copyright (c) 2020-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.


import React from 'react'
import {Provider as ReduxProvider} from 'react-redux'
import {render, screen, fireEvent, act} from '@testing-library/react'

import {mockDOM, wrapDNDIntl, mockStateStore} from '../../testUtils'
import {TestBlockFactory} from '../../test/testBlockFactory'

import Editor from './editor'

jest.mock('draft-js/lib/generateRandomKey', () => () => '123')

describe('components/blocksEditor/editor', () => {
    beforeEach(mockDOM)

    const board1 = TestBlockFactory.createBoard()
    board1.id = 'board-id-1'

    const state = {
        users: {
            boardUsers: {
                1: {username: 'abc'},
                2: {username: 'd'},
                3: {username: 'e'},
                4: {username: 'f'},
                5: {username: 'g'},
            },
        },
        boards: {
            current: 'board-id-1',
            boards: {
                [board1.id]: board1,
            },
        },
        clientConfig: {
            value: {},
        },
    }
    const store = mockStateStore([], state)

    test('should match snapshot', async () => {
        let container
        await act(async () => {
            const result = render(wrapDNDIntl(
                <ReduxProvider store={store}>
                    <Editor
                        id='block-id'
                        boardId='fake-board-id'
                        initialValue='test-value'
                        initialContentType='text'
                        onSave={jest.fn()}
                    />
                </ReduxProvider>,
            ))
            container = result.container
        })
        expect(container).toMatchSnapshot()
    })

    test('should match snapshot on empty', async () => {
        let container
        await act(async () => {
            const result = render(wrapDNDIntl(
                <ReduxProvider store={store}>
                    <Editor
                        boardId='fake-board-id'
                        onSave={jest.fn()}
                    />
                </ReduxProvider>,
            ))
            container = result.container
        })
        expect(container).toMatchSnapshot()
    })

    test('should call onSave after introduce text and hit enter', async () => {
        const onSave = jest.fn()
        await act(async () => {
            render(wrapDNDIntl(
                <ReduxProvider store={store}>
                    <Editor
                        boardId='fake-board-id'
                        onSave={onSave}
                    />
                </ReduxProvider>,
            ))
        })
        let input = screen.getByDisplayValue('')
        expect(onSave).not.toBeCalled()
        fireEvent.change(input, {target: {value: '/title'}})
        fireEvent.keyDown(input, {key: 'Enter'})
        expect(onSave).not.toBeCalled()

        input = screen.getByDisplayValue('')
        fireEvent.change(input, {target: {value: 'test'}})
        fireEvent.keyDown(input, {key: 'Enter'})

        expect(onSave).toBeCalledWith(expect.objectContaining({value: 'test'}))
    })
})
