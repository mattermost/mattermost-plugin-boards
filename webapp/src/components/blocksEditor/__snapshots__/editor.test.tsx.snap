// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/blocksEditor/editor should match snapshot 1`] = `
<div>
  <div
    class="Editor"
  >
    <div
      class="TextContent"
      data-testid="text"
    >
      <div
        class="MarkdownEditor octo-editor  active"
      >
        <div
          class="MarkdownEditorInput"
        >
          <div
            class="DraftEditor-root"
          >
            <div
              class="DraftEditor-editorContainer"
            >
              <div
                aria-autocomplete="list"
                aria-expanded="false"
                class="notranslate public-DraftEditor-content"
                contenteditable="true"
                role="combobox"
                spellcheck="false"
                style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
              >
                <div
                  data-contents="true"
                >
                  <div
                    class=""
                    data-block="true"
                    data-editor="123"
                    data-offset-key="123-0-0"
                  >
                    <div
                      class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                      data-offset-key="123-0-0"
                    >
                      <span
                        data-offset-key="123-0-0"
                      >
                        <span
                          data-text="true"
                        >
                          test-value
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/blocksEditor/editor should match snapshot on empty 1`] = `
<div>
  <div
    class="Editor"
  >
    <div
      class="RootInput css-b62m3t-container"
    >
      <span
        class="css-1f43avz-a11yText-A11yText"
        id="react-select-2-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="css-1f43avz-a11yText-A11yText"
      />
      <div
        class=" css-1haocjs-control"
      >
        <div
          class=" css-b2z5qd-ValueContainer"
        >
          <div
            class=" css-1jqq78o-placeholder"
            id="react-select-2-placeholder"
          >
            Introduce your text or your slash command
          </div>
          <div
            class=" css-26cneq-Input"
            data-value=""
          >
            <input
              aria-autocomplete="list"
              aria-describedby="react-select-2-placeholder"
              aria-expanded="false"
              aria-haspopup="true"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              class=""
              id="react-select-2-input"
              role="combobox"
              spellcheck="false"
              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        />
      </div>
    </div>
  </div>
</div>
`;
